src/cluster/kv/fake/store.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2020  Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package fake
+
+import (
+	"errors"
+
+	"github.com/m3db/m3/src/cluster/kv"
+
+	"github.com/golang/protobuf/proto"
+)
+
+// NewStore returns a fakeStore adhering to the kv.Store interface.
+// All methods except Watch are implemented. Implementation is not threadsafe
+// and should only be used for tests.
+func NewStore() kv.Store {
+	return &fakeStore{
+		store: make(map[string][]kv.Value),
+	}
+}
+
+type fakeStore struct {
+	store map[string][]kv.Value
+}
+
+func (f *fakeStore) Get(key string) (kv.Value, error) {
+	value, ok := f.store[key]
+	if !ok {
+		return nil, kv.ErrNotFound
+	}
+
+	return value[len(value)-1], nil
+}
+
+func (f *fakeStore) Watch(_ string) (kv.ValueWatch, error) {
+	panic("implement me")
```

Comment:
Leaving this for now?

Reply:
Yeah, I've got a comment at the top that `Watch` on the fake is left unimplemented. Want to leave this here just in case someone accidentally uses it.